knowledge-base/treelist-get-parent-item-on-insert.md

```diff
@@ -0,0 +1,312 @@
+    }
+
+    public List<Employee> Data { get; set; }
+    public Employee CurrentlyEditedEmployee { get; set; }
+
+    protected override async Task OnInitializedAsync()
+    {
+        Data = await GetTreeListData();
+    }
+
+    // sample model
+
+    public class Employee
+    {
+        // denote the parent-child relationship between items
+        public int Id { get; set; }
+        public int? ParentId { get; set; }
+
+        // custom data fields for display
+        public string Name { get; set; }
+        public string EmailAddress { get; set; }
+        public DateTime HireDate { get; set; }
+    }
+
+    // data generation
+
+    async Task<List<Employee>> GetTreeListData()
+    {
+        List<Employee> data = new List<Employee>();
+
+        for (int i = 1; i < 15; i++)
+        {
+            data.Add(new Employee
+            {
+                Id = i,
+                ParentId = null, // indicates a root-level item
+                Name = $"root: {i}",
+                EmailAddress = $"{i}@example.com",
+                HireDate = DateTime.Now.AddYears(-i)
+            }); ;
+
+            for (int j = 1; j < 4; j++)
+            {
+                int currId = i * 100 + j;
+                data.Add(new Employee
+                {
+                    Id = currId,
+                    ParentId = i,
+                    Name = $"first level child {j} of {i}",
+                    EmailAddress = $"{currId}@example.com",
+                    HireDate = DateTime.Now.AddDays(-currId)
+                });
+
+                for (int k = 1; k < 3; k++)
+                {
+                    int nestedId = currId * 1000 + k;
+                    data.Add(new Employee
+                    {
+                        Id = nestedId,
+                        ParentId = currId,
+                        Name = $"second level child {k} of {i} and {currId}",
+                        EmailAddress = $"{nestedId}@example.com",
+                        HireDate = DateTime.Now.AddMinutes(-nestedId)
+                    }); ;
+                }
+            }
+        }
+
+        return await Task.FromResult(data);
+    }
+}
+````
+
+### Get the Parent Item When Inserting a Child to the TreeList When Bound To Hierarchical Data
+
+#### Step by step Explanations
+
+1. Create a [custom command]({%slug treelist-columns-command%}) to initiate inserting in the `TreeListCommandColumn`.
+1. Pass the `TreeListCommandEventsArgs` object to the `OnClick` handler.
+1. Set the `ParentItem` field of the [TreeList state]({%slug treelist-state%}) to the item received from the `TreeListCommandEventsArgs`.
+1. Use the TreeList state and its `InsertedItem` field to populate the desired information from the current item that you have in the event arguments..
+
+
+````CSHTML
+@* One way to get metadata in the current item about its parent upon insertion *@
+
+<TelerikTreeList Data="@Data"
+                 EditMode="@TreeListEditMode.Popup"
```

Comment:
I added this, make sure things are OK with it and the editor template, also test with InLine mode.